### DISCUSSION TITLE
How can I programatically generate slides from a list of inputs?

### DISCUSSION THREAD

**Discussion by mtqoi** [quarto-dev/quarto-cli#6174] [Q&A] [answered]
### Description

I am happy generating slides "manually" with quarto - specifying by hand where to begin new slides and what each one should contain. The styling and the formatting are all fine for me. Primarily I work with either python Jupyter notebooks or `.qmd` files but I can learn new frameworks if needed

But I am interested in doing this in a more automated way from a list of inputs. 

Concrete use-case: I am running an analysis on several countries' data for my company, and want to create a slide deck to communicate the analysis. We have a list of 50+ countries and I would like each country on its own slide.

Ideally I want to do something like the following pseudocode:

```
for country in countries_list:
    generate_slide(country)
```

and the output should be the full slide deck, one slide for each country, plus possibly some one-off introductory slides that I can manually add

**Comment by cscheid** [comment, marked as answer]
I would generate the .qmd file programmatically in a python script and maybe run that in a [project prerender script](https://quarto.org/docs/projects/scripts.html#pre-and-post-render).

You can even factor this out nicely by keeping the manual portion of your slide deck constant in a `slides.qmd`, which would have an include statement for the auto-generated file. Something like this:


```
---
title: My slide deck
format: revealjs
---

## A slide deck

These are manual slides

{{< include _autogenerated.qmd >}}

```

Your python script would then generate the markdown file in `_autogenerated.qmd` however you wanted. See https://quarto.org/docs/authoring/includes.html.

**Reply by mtqoi** [reply]
Amazing, thanks @cscheid this is exactly what I was looking for

**Reply by robayo** [reply]
Will there be a function such as Rmarkdown asis=True to generate verbatim quarto text from executed code?

**Reply by mcanouil** [reply]
Not in Quarto as it is independent of the computing language.
Use knitr and nearly all its tooling and/or `output: asis` (if you know what you are doing).

**Reply by cderv** [reply]
Doc about `output: asis` to produce Raw output: https://quarto.org/docs/computations/execution-options.html#raw-output

`output: asis` is a Quarto feature to be clear. It will do the right thing regarding Jupyter and Knitr engine. As said, this is is then to the user to use the right tools and functions inside the chunk for the engine used.
